### PR TITLE
Remove obsolete methods from DiagnosticDescriptorBuilder

### PR DESCRIPTION
**Pull request by pavel-mikula-sonarsource**
Cleanup after #5590

### CODE REVIEW

File: analyzers/src/SonarAnalyzer.Common/DiagnosticAnalyzer/DiagnosticDescriptorBuilder.cs [hunk 1]
```diff
@@ -85,14 +67,6 @@ public static class DiagnosticDescriptorBuilder
...
 
-        [Obsolete]  // Will be removed with obsolete GetDescriptor method
-        private static string[] BuildTags(string diagnosticId, ResourceManager resourceManager, bool fadeOutCode) =>
-            BuildTags(
-                resourceManager.GetString("RoslynLanguage"),
-                bool.Parse(resourceManager.GetString($"{diagnosticId}_IsActivatedByDefault")),
-                (SourceScope)Enum.Parse(typeof(SourceScope), resourceManager.GetString($"{diagnosticId}_Scope")),
-                fadeOutCode);
-
         private static string[] BuildTags(string languageName, bool sonarWay, SourceScope scope, bool fadeOutCode)
```
Comment (csaba-sagi-sonarsource) [inline analyzers/src/SonarAnalyzer.Common/DiagnosticAnalyzer/DiagnosticDescriptorBuilder.cs:70]: Who knows when will be the next time to touch this class, I suggest to change `BuildUtilityTags` to expression bodied method.

File: analyzers/tests/SonarAnalyzer.UnitTest/Helpers/DiagnosticDescriptorBuilderTest.cs [hunk 2]
```diff
@@ -18,86 +18,13 @@
...
-
-            // Act
-            var result = DiagnosticDescriptorBuilder.GetDescriptor(diagnosticId, "", mockedResourceManager);
-
-            // Assert
-            result.CustomTags.Should().OnlyContain(DiagnosticDescriptorBuilder.MainSourceScopeTag, LanguageValue);
-        }
-
         [TestMethod]
         public void GetUtilityDescriptor_Should_Contain_NotConfigurable_CustomTag()
```
Comment (csaba-sagi-sonarsource) [inline analyzers/tests/SonarAnalyzer.UnitTest/Helpers/DiagnosticDescriptorBuilderTest.cs:29]: Nitpick: use `string.Empty` for empty string.

**Review by csaba-sagi-sonarsource — APPROVED** [review]
LGTM! I left two nitpick comments, that should be addressed.

File: analyzers/tests/SonarAnalyzer.UnitTest/Helpers/DiagnosticDescriptorBuilderTest.cs (hunk 2 above)
Comment (pavel-mikula-sonarsource) [reply analyzers/tests/SonarAnalyzer.UnitTest/Helpers/DiagnosticDescriptorBuilderTest.cs:29]: Done in #5858

**Pull request merged**